bsroot.sh: open /home/%username%/bsroot/html/static/ingress.yaml: no such file or directory

在执行 bsroot.sh 生成配置文件的时候（指定bsroot 目录，并且bsroot不在当前路径的情况下）：

``` shell
Preparing setup kubectl ... Done
Generating configuration files ...2016/10/28 02:13:03 open /home/%username%/bsroot/html/static/ingress.yaml: no such file or directory
panic: open /home/%username%/bsroot/html/static/ingress.yaml: no such file or directory

goroutine 1 [running]:
panic(0x6d47a0, 0xc42012e430)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
log.Panic(0xc42003de08, 0x1, 0x1)
	/usr/local/go/src/log/log.go:320 +0xc9
github.com/topicai/candy.Must(0x8a0480, 0xc42006de90)
	/go/src/github.com/topicai/candy/file.go:14 +0x8a
github.com/topicai/candy.mustCreate(0x7fff23ba5e76, 0x2d, 0xc4200889a0, 0xc42008a840)
	/go/src/github.com/topicai/candy/file.go:26 +0x57
github.com/topicai/candy.WithCreated(0x7fff23ba5e76, 0x2d, 0xc42003df20)
	/go/src/github.com/topicai/candy/file.go:56 +0x39
main.main()
	/go/src/github.com/k8sp/sextant/addons/addons.go:68 +0x23c
Failed to generate ingress.yaml !
```
解决：
在所有的 docker run 代码里面增加一个参数：
--volume $BSROOT_IN:$BSROOT_IN \

``` shell
    docker run --rm -it \
            --volume $GOPATH:/go \
            --volume $CLUSTER_DESC:$CLUSTER_DESC \
            --volume $BSROOT_IN:$BSROOT_IN \
            golang:wheezy \
            /go/bin/addons -cluster-desc-file $CLUSTER_DESC \
        -template-file $SEXTANT_DIR_IN/addons/template/ingress.template \
        -config-file $BSROOT_IN/html/static/ingress.yaml || \
        { echo 'Failed to generate ingress.yaml !' ; exit 1; }
```

--- a/bsroot.sh
+++ b/bsroot.sh
@@ -242,6 +242,7 @@ EOF
     cd $BSROOT/html/static/$VERSION
     gpg --verify coreos_production_image.bin.bz2.sig > /dev/null 2>&1 || { echo "Failed"; exit 1; }
     cd $BSROOT/html/static
+    rm -fr current
     ln -sf ./$VERSION current || { echo "Failed"; exit 1; }
     echo "Done"
 }
@@ -346,15 +347,7 @@ generate_addons_config() {
     docker run --rm -it \
             --volume $GOPATH:/go \
             --volume $CLUSTER_DESC:$CLUSTER_DESC \
-            golang:wheezy \
-            /go/bin/addons -cluster-desc-file $CLUSTER_DESC \
-            -template-file $SEXTANT_DIR_IN/addons/template/ingress.template \
-            -config-file $BSROOT_IN/html/static/ingress.yaml || \
-            { echo 'Failed to generate ingress.yaml !' ; exit 1; }
-
-    docker run --rm -it \
-            --volume $GOPATH:/go \
-            --volume $CLUSTER_DESC:$CLUSTER_DESC \
+            --volume $BSROOT_IN:$BSROOT_IN \
             golang:wheezy \
             /go/bin/addons -cluster-desc-file $CLUSTER_DESC \
         -template-file $SEXTANT_DIR_IN/addons/template/ingress.template \
@@ -364,6 +357,7 @@ generate_addons_config() {
     docker run --rm -it \
             --volume $GOPATH:/go \
             --volume $CLUSTER_DESC:$CLUSTER_DESC \
+            --volume $BSROOT_IN:$BSROOT_IN \
             golang:wheezy \
             /go/bin/addons -cluster-desc-file $CLUSTER_DESC \
         -template-file $SEXTANT_DIR_IN/addons/template/skydns.template \
@@ -373,6 +367,7 @@ generate_addons_config() {
     docker run --rm -it \
             --volume $GOPATH:/go \
             --volume $CLUSTER_DESC:$CLUSTER_DESC \
+            --volume $BSROOT_IN:$BSROOT_IN \
             golang:wheezy \
             /go/bin/addons -cluster-desc-file $CLUSTER_DESC \
         -template-file $SEXTANT_DIR_IN/addons/template/skydns-service.template \
@@ -382,6 +377,7 @@ generate_addons_config() {
     docker run --rm -it \
             --volume $GOPATH:/go \
             --volume $CLUSTER_DESC:$CLUSTER_DESC \
+            --volume $BSROOT_IN:$BSROOT_IN \
             golang:wheezy \
             /go/bin/addons -cluster-desc-file $CLUSTER_DESC \
         -template-file $SEXTANT_DIR_IN/addons/template/dnsmasq.conf.template \

--- a/bsroot.sh
+++ b/bsroot.sh
@@ -242,7 +242,8 @@ EOF
     cd $BSROOT/html/static/$VERSION
     gpg --verify coreos_production_image.bin.bz2.sig > /dev/null 2>&1 || { echo "Failed"; exit 1; }
     cd $BSROOT/html/static
-    rm -fr current
+    # Never change 'current' to 'current/', I beg you. 
+    rm -rf current > /dev/null 2>&1 
     ln -sf ./$VERSION current || { echo "Failed"; exit 1; }
     echo "Done"
 }

